CI/Dockerfile

```diff
@@ -250,28 +237,44 @@ RUN mkdir -p ${dagmc_build_dir}/build && \
         -DCMAKE_C_COMPILER=${CC} \
         -DCMAKE_CXX_COMPILER=${CXX} \
         -DCMAKE_Fortran_COMPILER=gfortran \
-        -DCMAKE_INSTALL_PREFIX=${dagmc_install_dir} \
+        -DCMAKE_INSTALL_PREFIX=${DAGMC_INSTALL_DIR} \
         -DDOUBLE_DOWN=$([ "${DOUBLE_DOWN_VERSION}" != "off" ] && echo "ON" || echo "OFF") \
-        -Ddd_ROOT=${double_down_install_dir} && \
+        -Ddd_ROOT=${DOUBLE_DOWN_INSTALL_DIR} && \
     make -j${CI_JOBS} && \
     make install
 
 
-FROM dagmc as dagmc_test
+FROM dagmc AS dagmc_test
 
-ENV dagmc_build_dir=${build_dir}/dagmc
-ENV dagmc_install_dir=${install_dir}/dagmc
+ENV DAGMC_BUILD_DIR=${BUILD_DIR}/dagmc
+ENV DAGMC_INSTALL_DIR=${INSTALL_DIR}/dagmc
 
 # Test DAGMC executables
-RUN cd ${dagmc_build_dir}/build && \
-    PATH=${dagmc_install_dir}/bin:$PATH CTEST_OUTPUT_ON_FAILURE=1 make test
+RUN cd ${DAGMC_BUILD_DIR}/build && \
+    PATH=${DAGMC_INSTALL_DIR}/bin:$PATH CTEST_OUTPUT_ON_FAILURE=1 make test
 
 # clean out config test directory for next build
-RUN cd ${dagmc_build_dir} && \
+RUN cd ${DAGMC_BUILD_DIR} && \
     git clean -dxf .
 
 # Test DAGMC CMake configuration file
-RUN cd ${dagmc_build_dir}/cmake/test_config && \
-    cmake . -DDAGMC_ROOT=${dagmc_install_dir} && \
+RUN cd ${DAGMC_BUILD_DIR}/cmake/test_config && \
+    cmake . -DDAGMC_ROOT=${DAGMC_INSTALL_DIR} && \
     CTEST_OUTPUT_ON_FAILURE=1 make all test
 
+FROM scratch AS binaries
+
+ARG INSTALL_DIR
+
+COPY --from=dagmc_test ${INSTALL_DIR} ${INSTALL_DIR}
```

Comment:
What files are you copying here and why?  If they aren't installed by `make install` then they probably aren't necessary.

Reply:
In this step, I am using an empty image to copy all the installed files from the DAGMC step to the exact location. Therefore, in this image, we only contain the files available in /opt/*. There are no extra files or dependencies in this new image, which keeps the image size minimal, less than 90MB.

Reply:
In this step we have, DAGMC, MOAB, HDF5, Double Down, Embree, Geant4. However, We can make it more minimal by copying only `/opt/dagmc` folder. I have chosen to copy the full folder as I am not sure if they are core dependency of DAGMC or only necessary to build DAGMC since we have all the binaries available that works without them.

Reply:
By applying this step, we can update our docker_publish workflow to add a new tag as `dagmc:latest` upon PR merge, and `dagmc:version/stable` upon new release. This can help streamline other projects as well, such as PyNE and NukeLab. We can also update our documentations to use the docker version of DAGMC as well for the general users. Let me know if you find the idea useful.

Reply:
Ah! You've already merged this PR. I was waiting for your response. The binary stage won't run until we update the docker_publish file. There are currently no instructions for this. I think I should create a new PR for it.